src/puppetlabs/puppetdb/time.clj

```diff
@@ -201,3 +204,36 @@
 (def to-millis
   "Given an instance of `Period`, return an integer representing the number of milliseconds"
   (partial to-unit #(.getMillis %)))
+
+(def ordered-formatters
+  "Like clj-time.format/formatters but ordered with the more likely
+  successful formats first"
+  (let [likely-formats [:date-time :date :date-time-no-ms :basic-date :basic-date-time]]
+    (map tf/formatters (concat likely-formats
+                               (remove (set likely-formats) (keys tf/formatters))))))
+
+(s/defn ^:always-validate attempt-date-time-parse
+  "Parses `timestamp-str` using `formatter`. Returns nil if an
+  Exception is thrown, otherwise the parsed DateTime object"
+  [formatter :- DateTimeFormatter
+   timestamp-str :- String]
+  (try
+    (tf/parse formatter timestamp-str)
+    (catch Exception _ nil)))
```

Comment:
Wondered if it could make any sense to try to catch a more specific set of exceptions here, unless we really do want to keep going no matter what (and maybe we do), or unless we don't really know/care what tf/parse might throw.

I'm just thinking (generally) about situations where something might be more seriously wrong, but we have to wade through all the parsers before we hear about it.